core/tx_pool.go

```diff
@@ -272,6 +279,9 @@ type TxPool struct {
 	initDoneCh      chan struct{}  // is closed once the pool is initialized (for tests)
 
 	changesSinceReorg int // A counter for how many drops we've performed in-between reorg.
+
+	enableTxLifecycles bool
+	txLifecycles       *lru.ARCCache // Snapshots for recent block to speed up reorgs
```

Comment:
why is this an LRU cache rather than simply a map?

Reply:
Because just open this metrics on rpc nodes, some tx maybe don't broadcast to rpc nodes, if just put in the map, the map maybe becomes bigger and bigger. 

It's better have a TTL map to do it, but I don't want import a new map (like: https://github.com/patrickmn/go-cache). I found LRU map can also do this thing.

Reply:
> if just put in the map, the map maybe becomes bigger and bigger.

that shouldn't be a problem if we make the map follow the eviction rules of the txpool itself. So, if a txn is being removed from the pool for whatever reason, we should remove it's arrival time from the pool as well.

If we use a LRU cache, the entries we put in it might get evicted at any point, making our measurements less accurate.

Reply:
Doesnt look like we need this at all, `types.Transaction` already has a `time` field

Reply:
if the map doesn't have the expiration function, there will be a case.
* because `calculate tx lifecycle feature` only enables on follower nodes, like bootnode, rpc, internal-rpc, some tx maybe can't broadcast to these nodes. if there are many tx don't broadcast to nodes, the key/value doesn't be evicted, the map will become large.

Reply:
combined with @HAOYUatHZ 's concern (separating "dropped case" and "included case"), maybe we can:
1. based on Omer's suggestion, reuse this field (`Time first seen locally`, not sure if it fits the need in special cases cc @omerfirmak, e.g. the transaction is dropped then resubmitted to the same node): https://github.com/scroll-tech/go-ethereum/blob/scroll-v5.6.0/core/types/transaction.go#L62
2. submit two metrics, one for "included", and another for "dropped", then won't affect the p50/p90/p99 time to be included.

Reply:
fixed [96dd0c2](https://github.com/scroll-tech/go-ethereum/pull/839/commits/96dd0c284a656f72251875ce93ea5931f6312d46)